### PR TITLE
native_group_norm_backward: Fix precision errors

### LINKED ISSUE #180740: [Inductor] functorch_dp_cifar10 accuracy regression in dynamic_inductor_torchbench
### Description

`functorch_dp_cifar10` fails accuracy checks in the `dynamic_inductor_torchbench` periodic CI job (CUDA, A10G):

```
functorch_dp_cifar10    FAIL:     accuracy=fail_accuracy, expected=pass
```

The error is a numerical accuracy regression in batch norm parameters:
```
Accuracy failed for key name layer2.0.bn2.bias
RMSE (res-fp64): 0.00606, (ref-fp64): 0.00145, shape=torch.Size([128])
```

### Details

- Only affects `dynamic_inductor_torchbench` (dynamic shapes + inductor), not static shapes
- Consistent across at least 4 consecutive periodic CI runs (April 17-18)
- Inference pass succeeds; only training accuracy fails
- The model compiles successfully (1 graph, 69 ops, 0 graph breaks)

### Expected CSV entry
`benchmarks/dynamo/ci_expected_accuracy/dynamic_inductor_torchbench_inference.csv`: `functorch_dp_cifar10,pass,0`

### CI Impact

Blocks `periodic-dynamo-benchmarks-test / test (dynamic_inductor_torchbench, 1, 2, linux.g5.4xlarge.nvidia.gpu)`.

cc @ezyang @penguinwu @bobrenjc93 @aditvenk @laithsakka @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### PR DESCRIPTION
**Pull request by benjaminglass1**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.16.0) (oldest at bottom):
* __->__ #190245

Fixes missed upcasts in the CUDA kernel and decomposition for native_group_norm_backward.  Re-works infinitely_differentiable_native_group_norm_backward to use native_group_norm_backward where possible (ensures we actually use the decomposition in cases where Inductor can't prove the other two gradients are unused).  Fixes a broken Inductor benchmark in the pipeline, as well as three XFAILed tests.

Closes #180740

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

**Comment by jansel** [comment]
@claude review these changes

**Comment by slayton58** [comment]
This looks good to me, I think the suggested comment change from janselbot's review is valid, and would be worth quickly doing, but otherwise nothing else I can see to mention.

File: test/inductor/test_torchinductor_opinfo.py [hunk 1]
```diff
@@ -569,9 +569,6 @@ def wrapper_noop_set_seed(op, *args, **kwargs):
         "atol": 1e-4,
         "rtol": 7e-1,
     },
-    # The eager gradient for native_group_norm appears to be numerically unstable at low
-    # precisions; more investigation is needed.
-    ("native_group_norm", f16): {"check_gradient": False},
```
Comment (slayton58) [inline test/inductor/test_torchinductor_opinfo.py:574]: This is scary

File: test/inductor/test_torchinductor_opinfo.py [hunk 2]
```diff
@@ -739,9 +736,6 @@ def wrapper_noop_set_seed(op, *args, **kwargs):
     ("nn.functional.interpolate.trilinear", f64): {
         "check_gradient": False,
     },
-    # The eager gradient for native_group_norm appears to be numerically unstable at low
-    # precisions; more investigation is needed.
-    ("native_group_norm", f16): {"check_gradient": False},
```
Comment (slayton58) [inline test/inductor/test_torchinductor_opinfo.py:744]: Same here

**Review by slayton58 — COMMENTED** [review]
~These need to be tracked down before merge~

I need more coffee, nothing to be changed here

File: test/inductor/test_torchinductor_opinfo.py (hunk 1 above)
Comment (benjaminglass1) [reply test/inductor/test_torchinductor_opinfo.py:574]: @slayton58 I'm not super scared by them, because I put them there. :sweat_smile: They were from earlier PRs in the stack, before you got involved, as I was expanding `native_group_norm` test coverage but before I found the numerical issues.

Comment (slayton58) [reply test/inductor/test_torchinductor_opinfo.py:574]: Yep, I misread this completely, you're *removing* the sketchiness

File: test/inductor/test_torchinductor_opinfo.py (hunk 2 above)
Comment (slayton58) [reply test/inductor/test_torchinductor_opinfo.py:744]: Ignore, same reason (dumbness on my end) as above

**Review by slayton58 — APPROVED** [review]
